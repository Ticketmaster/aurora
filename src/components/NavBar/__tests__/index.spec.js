import React from "react";
import renderer from "react-test-renderer";

import NavBar from "../index";

describe("NavBar", () => {
  it("renders with defaults", () =>
    expect(
      renderer
        .create(
          <NavBar>
            <NavBar.MenuButton isFirst />
            Content
            <NavBar.LinkRow>
              <NavBar.Link>Link 1</NavBar.Link>
              <NavBar.Link>Link 2</NavBar.Link>
              <NavBar.Link>Link 3</NavBar.Link>
              <NavBar.Link>
                Link 4
                <NavBar.LinkList style={{ minWidth: "258px" }}>
                  <NavBar.Link>Link 5</NavBar.Link>
                  <NavBar.Link>Link 6</NavBar.Link>
                  <NavBar.Link>Link 7 With Really Long Content </NavBar.Link>
                </NavBar.LinkList>
              </NavBar.Link>
            </NavBar.LinkRow>
            <NavBar.SearchButton />
            <NavBar.UserButton />
            <NavBar.TextButton isLast>Text Button!</NavBar.TextButton>
          </NavBar>
        )
        .toJSON()
    ).toMatchSnapshot());

  it("renders fixed", () =>
    expect(
      renderer
        .create(
          <NavBar fixed>
            <NavBar.MenuButton isFirst />
            Content
            <NavBar.LogoContainer href="http://localhost/new/">
              Logo
            </NavBar.LogoContainer>
            <NavBar.LinkRow>
              <NavBar.Link href="/">Link 1</NavBar.Link>
              <NavBar.Link>Link 2</NavBar.Link>
              <NavBar.Link>Link 3</NavBar.Link>
              <NavBar.Link>
                Link 4
                <NavBar.LinkList style={{ minWidth: "258px" }}>
                  <NavBar.Link>Link 5</NavBar.Link>
                  <NavBar.Link>Link 6</NavBar.Link>
                  <NavBar.Link>Link 7 With Really Long Content </NavBar.Link>
                </NavBar.LinkList>
              </NavBar.Link>
            </NavBar.LinkRow>
          </NavBar>
        )
        .toJSON()
    ).toMatchSnapshot());

  it("renders inverted", () =>
    expect(
      renderer
        .create(
          <NavBar invert>
            <NavBar.MenuButton isFirst invert />
            <NavBar.LogoContainer href="http://localhost/new/">
              Logo
            </NavBar.LogoContainer>
            <NavBar.LinkRow>
              <NavBar.Link>Link 1</NavBar.Link>
              <NavBar.Link>Link 2</NavBar.Link>
              <NavBar.Link>Link 3</NavBar.Link>
              <NavBar.Link>
                Link 4
                <NavBar.LinkList style={{ minWidth: "258px" }}>
                  <NavBar.Link>Link 5</NavBar.Link>
                  <NavBar.Link>Link 6</NavBar.Link>
                  <NavBar.Link>Link 7 With Really Long Content </NavBar.Link>
                </NavBar.LinkList>
              </NavBar.Link>
            </NavBar.LinkRow>
          </NavBar>
        )
        .toJSON()
    ).toMatchSnapshot());
});
