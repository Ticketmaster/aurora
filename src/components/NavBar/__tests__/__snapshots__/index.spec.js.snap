// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders fixed 1`] = `
.c10 {
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  width: 205px;
  overflow: hidden;
  background-color: rgba(255,255,255,1);
  padding: 8px;
  border-radius: 2px;
}

.list-container--open .c10 {
  display: block;
}

.c10.links__list--open {
  display: block;
}

.c7 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  border: 0;
}

.links__list .c7 {
  width: 100%;
  text-align: left;
}

.c7:focus,
.c7:hover {
  outline: 0;
}

.c8 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  border: 0;
}

.links__list .c8 {
  width: 100%;
  text-align: left;
}

.c8:focus,
.c8:hover {
  outline: 0;
}

.c5 {
  height: 60px;
}

.c5 .list-container {
  position: relative;
}

.c9 {
  top: 60px;
  left: -74%;
}

.c6 {
  background-color: transparent;
  padding: 19px 16px 19px 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c6,
.links__list .c6 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c6:visited,
.c6:active {
  color: rgba(255,255,255,1);
}

.nav--inverted .c6:visited,
.nav--inverted .c6:active,
.links__list .c6:visited,
.links__list .c6:active {
  color: rgba(38,38,38,1) !important;
}

.c6:focus,
.c6:hover {
  outline: 0;
  background-color: rgba(38,38,38,0.1);
}

.nav--inverted .c6:focus,
.nav--inverted .c6:hover,
.links__list .c6:focus,
.links__list .c6:hover {
  background-color: #E6E6E6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c2 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 8px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c4 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c3 {
  padding-left: 8px;
  padding-right: 8px;
}

.c0 {
  top: 0;
  left: 0;
  right: 0;
  min-height: 60px;
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  color: rgba(255,255,255,1);
  font-weight: 600;
}

.c0.nav--relative {
  position: relative;
}

.c0.nav--absolute {
  position: absolute;
}

.c0.nav--fixed {
  position: fixed;
}

.c0 .linkItem:visited {
  color: rgba(255,255,255,1);
}

.c0.nav--inverted {
  background-color: rgba(255,255,255,1);
  border-bottom: 1px solid #D1D1D1;
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c0.nav--inverted .linkItem:visited {
  color: rgba(38,38,38,1);
}

.c0.nav--overlay:after {
  content: "";
  opacity: 0.4;
  top: 0;
  left: 0;
  right: 0;
  position: absolute;
  height: 60px;
  z-index: 4;
  background-image: linear-gradient(77deg,rgba(0,0,0,0),#000000);
}

.c0.nav--fade-in {
  -webkit-animation: fadeIn 0.3s ease-in-out;
  animation: fadeIn 0.3s ease-in-out;
}

.c0.nav--fade-out {
  -webkit-animation: fadeOut 0.3s ease-in-out;
  animation: fadeOut 0.3s ease-in-out;
}

.c1 {
  width: 100%;
  height: 60px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  z-index: 5;
}

@media screen and (min-width:480px) {
  .c2 {
    padding-right: 12px;
    padding-left: 24px;
  }
}

@media screen and (min-width:480px) {
  .c4 {
    padding-right: 12px;
    padding-left: 12px;
  }
}

@media screen and (min-width:480px) {
  .c3 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    padding-left: 24px;
  }
}

@media screen and (min-width:480px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(82deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(86deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c1 {
    margin: 0 auto;
  }
}

@media (min-width:1440px) {
  .c1 {
    padding: 0 44px;
  }
}

<nav
  className="nav--fixed nav--fade-in c0"
  role="navigation"
  style={
    Object {
      "backgroundColor": null,
    }
  }
>
  <div
    className="c1"
  >
    <button
      className="c2"
      type="button"
    >
      <svg
        fill="currentcolor"
        height={20}
        viewBox="0 0 20 17"
        width={20}
      >
        <path
          d="M 0 0 H 20 V 3 H 0 z M 0 7 H 20 V 10 H 0 z M 0 14 H 20 V 17 H 0 z"
        />
      </svg>
    </button>
    Content
    <a
      className="c3 c4"
      href="http://localhost/new/"
      rel={null}
      target="_self"
    >
      Logo
    </a>
    <div
      className="links links__row c5"
      role="menubar"
    >
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <a
          className="c6 link c7"
          href="/"
          rel={null}
          role="link"
          target="_self"
        >
          Link 1
        </a>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c6 link c8"
          href={null}
          role="button"
        >
          Link 2
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c6 link c8"
          href={null}
          role="button"
        >
          Link 3
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="c6 link c8"
          href={null}
          role="button"
        >
          Link 4
        </button>
        <div
          className="c9 links links__list c10"
          role="menu"
          style={
            Object {
              "minWidth": "258px",
            }
          }
        >
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c8"
              href={null}
              role="button"
            >
              Link 5
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c8"
              href={null}
              role="button"
            >
              Link 6
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c8"
              href={null}
              role="button"
            >
              Link 7 With Really Long Content 
            </button>
          </span>
        </div>
      </span>
    </div>
  </div>
</nav>
`;

exports[`NavBar renders inverted 1`] = `
.c9 {
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  width: 205px;
  overflow: hidden;
  background-color: rgba(255,255,255,1);
  padding: 8px;
  border-radius: 2px;
}

.list-container--open .c9 {
  display: block;
}

.c9.links__list--open {
  display: block;
}

.c7 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  border: 0;
}

.links__list .c7 {
  width: 100%;
  text-align: left;
}

.c7:focus,
.c7:hover {
  outline: 0;
}

.c5 {
  height: 60px;
}

.c5 .list-container {
  position: relative;
}

.c8 {
  top: 60px;
  left: -74%;
}

.c6 {
  background-color: transparent;
  padding: 19px 16px 19px 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c6,
.links__list .c6 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c6:visited,
.c6:active {
  color: rgba(255,255,255,1);
}

.nav--inverted .c6:visited,
.nav--inverted .c6:active,
.links__list .c6:visited,
.links__list .c6:active {
  color: rgba(38,38,38,1) !important;
}

.c6:focus,
.c6:hover {
  outline: 0;
  background-color: rgba(38,38,38,0.1);
}

.nav--inverted .c6:focus,
.nav--inverted .c6:hover,
.links__list .c6:focus,
.links__list .c6:hover {
  background-color: #E6E6E6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c2 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 8px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c4 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c3 {
  padding-left: 8px;
  padding-right: 8px;
}

.c0 {
  top: 0;
  left: 0;
  right: 0;
  min-height: 60px;
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  color: rgba(255,255,255,1);
  font-weight: 600;
}

.c0.nav--relative {
  position: relative;
}

.c0.nav--absolute {
  position: absolute;
}

.c0.nav--fixed {
  position: fixed;
}

.c0 .linkItem:visited {
  color: rgba(255,255,255,1);
}

.c0.nav--inverted {
  background-color: rgba(255,255,255,1);
  border-bottom: 1px solid #D1D1D1;
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c0.nav--inverted .linkItem:visited {
  color: rgba(38,38,38,1);
}

.c0.nav--overlay:after {
  content: "";
  opacity: 0.4;
  top: 0;
  left: 0;
  right: 0;
  position: absolute;
  height: 60px;
  z-index: 4;
  background-image: linear-gradient(77deg,rgba(0,0,0,0),#000000);
}

.c0.nav--fade-in {
  -webkit-animation: fadeIn 0.3s ease-in-out;
  animation: fadeIn 0.3s ease-in-out;
}

.c0.nav--fade-out {
  -webkit-animation: fadeOut 0.3s ease-in-out;
  animation: fadeOut 0.3s ease-in-out;
}

.c1 {
  width: 100%;
  height: 60px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  z-index: 5;
}

@media screen and (min-width:480px) {
  .c2 {
    padding-right: 12px;
    padding-left: 24px;
  }
}

@media screen and (min-width:480px) {
  .c4 {
    padding-right: 12px;
    padding-left: 12px;
  }
}

@media screen and (min-width:480px) {
  .c3 {
    padding-left: 12px;
    padding-right: 12px;
  }
}

@media screen and (min-width:768px) {
  .c3 {
    padding-left: 24px;
  }
}

@media screen and (min-width:480px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(82deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(86deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c1 {
    margin: 0 auto;
  }
}

@media (min-width:1440px) {
  .c1 {
    padding: 0 44px;
  }
}

<nav
  className="nav--relative nav--inverted nav--fade-in c0"
  role="navigation"
  style={
    Object {
      "backgroundColor": null,
    }
  }
>
  <div
    className="c1"
  >
    <button
      className="c2"
      type="button"
    >
      <svg
        fill="currentcolor"
        height={20}
        viewBox="0 0 20 17"
        width={20}
      >
        <path
          d="M 0 0 H 20 V 3 H 0 z M 0 7 H 20 V 10 H 0 z M 0 14 H 20 V 17 H 0 z"
        />
      </svg>
    </button>
    <a
      className="c3 c4"
      href="http://localhost/new/"
      rel={null}
      target="_self"
    >
      Logo
    </a>
    <div
      className="links links__row c5"
      role="menubar"
    >
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c6 link c7"
          href={null}
          role="button"
        >
          Link 1
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c6 link c7"
          href={null}
          role="button"
        >
          Link 2
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c6 link c7"
          href={null}
          role="button"
        >
          Link 3
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="c6 link c7"
          href={null}
          role="button"
        >
          Link 4
        </button>
        <div
          className="c8 links links__list c9"
          role="menu"
          style={
            Object {
              "minWidth": "258px",
            }
          }
        >
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c7"
              href={null}
              role="button"
            >
              Link 5
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c7"
              href={null}
              role="button"
            >
              Link 6
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c6 link c7"
              href={null}
              role="button"
            >
              Link 7 With Really Long Content 
            </button>
          </span>
        </div>
      </span>
    </div>
  </div>
</nav>
`;

exports[`NavBar renders with defaults 1`] = `
.c7 {
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: absolute;
  width: 205px;
  overflow: hidden;
  background-color: rgba(255,255,255,1);
  padding: 8px;
  border-radius: 2px;
}

.list-container--open .c7 {
  display: block;
}

.c7.links__list--open {
  display: block;
}

.c5 {
  display: inline-block;
  -webkit-text-decoration: none;
  text-decoration: none;
  outline: 0;
  border: 0;
}

.links__list .c5 {
  width: 100%;
  text-align: left;
}

.c5:focus,
.c5:hover {
  outline: 0;
}

.c3 {
  height: 60px;
}

.c3 .list-container {
  position: relative;
}

.c6 {
  top: 60px;
  left: -74%;
}

.c4 {
  background-color: transparent;
  padding: 19px 16px 19px 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c4,
.links__list .c4 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c4:visited,
.c4:active {
  color: rgba(255,255,255,1);
}

.nav--inverted .c4:visited,
.nav--inverted .c4:active,
.links__list .c4:visited,
.links__list .c4:active {
  color: rgba(38,38,38,1) !important;
}

.c4:focus,
.c4:hover {
  outline: 0;
  background-color: rgba(38,38,38,0.1);
}

.nav--inverted .c4:focus,
.nav--inverted .c4:hover,
.links__list .c4:focus,
.links__list .c4:hover {
  background-color: #E6E6E6;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 16px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c2 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 8px;
  padding-left: 8px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c8 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 0 1 auto;
  -ms-flex: 0 1 auto;
  flex: 0 1 auto;
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  border: 0;
  height: 60px;
  outline: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding-right: 16px;
  padding-left: 8px;
  color: rgba(255,255,255,1);
  font-size: 16px;
  font-weight: 600;
}

.nav--inverted .c9 {
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c0 {
  top: 0;
  left: 0;
  right: 0;
  min-height: 60px;
  width: 100%;
  font-size: 16px;
  background-color: transparent;
  color: rgba(255,255,255,1);
  font-weight: 600;
}

.c0.nav--relative {
  position: relative;
}

.c0.nav--absolute {
  position: absolute;
}

.c0.nav--fixed {
  position: fixed;
}

.c0 .linkItem:visited {
  color: rgba(255,255,255,1);
}

.c0.nav--inverted {
  background-color: rgba(255,255,255,1);
  border-bottom: 1px solid #D1D1D1;
  color: rgba(38,38,38,1);
  font-weight: 400;
}

.c0.nav--inverted .linkItem:visited {
  color: rgba(38,38,38,1);
}

.c0.nav--overlay:after {
  content: "";
  opacity: 0.4;
  top: 0;
  left: 0;
  right: 0;
  position: absolute;
  height: 60px;
  z-index: 4;
  background-image: linear-gradient(77deg,rgba(0,0,0,0),#000000);
}

.c0.nav--fade-in {
  -webkit-animation: fadeIn 0.3s ease-in-out;
  animation: fadeIn 0.3s ease-in-out;
}

.c0.nav--fade-out {
  -webkit-animation: fadeOut 0.3s ease-in-out;
  animation: fadeOut 0.3s ease-in-out;
}

.c1 {
  width: 100%;
  height: 60px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  z-index: 5;
}

@media screen and (min-width:480px) {
  .c2 {
    padding-right: 12px;
    padding-left: 24px;
  }
}

@media screen and (min-width:480px) {
  .c8 {
    padding-right: 12px;
    padding-left: 12px;
  }
}

@media screen and (min-width:480px) {
  .c9 {
    padding-right: 24px;
    padding-left: 12px;
  }
}

@media screen and (min-width:480px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(82deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c0.nav--overlay:after {
    background-image: linear-gradient(86deg,rgba(0,0,0,0),#000000);
  }
}

@media screen and (min-width:768px) {
  .c1 {
    margin: 0 auto;
  }
}

@media (min-width:1440px) {
  .c1 {
    padding: 0 44px;
  }
}

<nav
  className="nav--relative nav--fade-in c0"
  role="navigation"
  style={
    Object {
      "backgroundColor": null,
    }
  }
>
  <div
    className="c1"
  >
    <button
      className="c2"
      type="button"
    >
      <svg
        fill="currentcolor"
        height={20}
        viewBox="0 0 20 17"
        width={20}
      >
        <path
          d="M 0 0 H 20 V 3 H 0 z M 0 7 H 20 V 10 H 0 z M 0 14 H 20 V 17 H 0 z"
        />
      </svg>
    </button>
    Content
    <div
      className="links links__row c3"
      role="menubar"
    >
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c4 link c5"
          href={null}
          role="button"
        >
          Link 1
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c4 link c5"
          href={null}
          role="button"
        >
          Link 2
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          className="c4 link c5"
          href={null}
          role="button"
        >
          Link 3
        </button>
      </span>
      <span
        className="list-container list-container--closed"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onTouchStart={[Function]}
        role="none"
      >
        <button
          aria-expanded={false}
          aria-haspopup={true}
          className="c4 link c5"
          href={null}
          role="button"
        >
          Link 4
        </button>
        <div
          className="c6 links links__list c7"
          role="menu"
          style={
            Object {
              "minWidth": "258px",
            }
          }
        >
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c4 link c5"
              href={null}
              role="button"
            >
              Link 5
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c4 link c5"
              href={null}
              role="button"
            >
              Link 6
            </button>
          </span>
          <span
            className="list-container list-container--closed"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onTouchStart={[Function]}
            role="none"
          >
            <button
              className="c4 link c5"
              href={null}
              role="button"
            >
              Link 7 With Really Long Content 
            </button>
          </span>
        </div>
      </span>
    </div>
    <button
      className="c8"
      type="button"
    >
      <svg
        fill="currentColor"
        height={21}
        viewBox="0 0 16 16"
        width={21}
      >
        <title>
          Search Icon
        </title>
        <path
          d="M11.491 6.048c0 .672-.135 1.31-.404 1.897a5.01 5.01 0 0 1-1.102 1.552 5.273 5.273 0 0 1-1.632 1.05 5.39 5.39 0 0 1-3.987 0 5.283 5.283 0 0 1-1.632-1.05 5.01 5.01 0 0 1-1.102-1.552 4.531 4.531 0 0 1-.404-1.897c0-.666.135-1.302.404-1.892.269-.592.64-1.114 1.102-1.552A5.304 5.304 0 0 1 4.366 1.56a5.207 5.207 0 0 1 1.994-.384c.706 0 1.376.13 1.993.384.62.256 1.17.607 1.632 1.045.462.438.833.96 1.102 1.552a4.54 4.54 0 0 1 .404 1.892m4.346 8.977l-4.744-4.93a6.36 6.36 0 0 0 .62-.774 5.913 5.913 0 0 0 .89-2.115 5.893 5.893 0 0 0-.004-2.346 5.543 5.543 0 0 0-.361-1.128 5.98 5.98 0 0 0-.588-1.036 6.49 6.49 0 0 0-.794-.925A6.37 6.37 0 0 0 8.794.461 6.82 6.82 0 0 0 7.61.116a6.742 6.742 0 0 0-2.5 0A6.82 6.82 0 0 0 3.925.46a6.44 6.44 0 0 0-2.064 1.313 6.491 6.491 0 0 0-.793.923 5.98 5.98 0 0 0-.588 1.036c-.16.361-.28.741-.36 1.128a5.954 5.954 0 0 0 0 2.376c.08.388.2.768.36 1.13a6.182 6.182 0 0 0 1.381 1.959c.298.288.624.544.97.764a6.44 6.44 0 0 0 2.279.894 6.77 6.77 0 0 0 3.363-.224 6.296 6.296 0 0 0 1.708-.873l4.749 4.925a.56.56 0 0 0 .215.145.701.701 0 0 0 .453.011.514.514 0 0 0 .2-.114.561.561 0 0 0 .039-.828"
        />
      </svg>
    </button>
    <button
      className="c8"
      type="button"
    >
      <svg
        height="24"
        size={24}
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fillRule="evenodd"
        >
          <path
            d="M24 0H0v24h24z"
          />
          <path
            d="M12 23C5.925 23 1 18.075 1 12S5.925 1 12 1s11 4.925 11 11-4.925 11-11 11zm0-1c5.523 0 10-4.477 10-10S17.523 2 12 2 2 6.477 2 12s4.477 10 10 10zm0-8a4 4 0 1 1 0-8 4 4 0 0 1 0 8zm0-1a3 3 0 1 0 0-6 3 3 0 0 0 0 6zm-5 7.5a.5.5 0 1 1-1 0V19a6 6 0 1 1 12 0v1.5a.5.5 0 1 1-1 0V19a5 5 0 0 0-10 0v1.5z"
            fill="#262626"
            fillOpacity=".5"
            fillRule="nonzero"
          />
        </g>
      </svg>
    </button>
    <button
      className="c9"
      type="button"
    >
      Text Button!
    </button>
  </div>
</nav>
`;
